tests/lib/Repository/Validator/TargetContentValidatorTest.php

```diff
@@ -0,0 +1,112 @@
+{
+    /** @var \eZ\Publish\API\Repository\ContentService|\PHPUnit_Framework_MockObject_MockObject */
+    private $contentService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService|\PHPUnit_Framework_MockObject_MockObject */
+    private $contentTypeService;
+
+    /** @var \Ibexa\Core\Repository\Validator\TargetContentValidator */
+    private $targetContentValidator;
+
+    public function setUp(): void
+    {
+        $this->contentService = $this->createMock(ContentService::class);
+        $this->contentTypeService = $this->createMock(ContentTypeService::class);
+
+        $this->targetContentValidator = new TargetContentValidator(
+            $this->contentService,
+            $this->contentTypeService
+        );
+    }
+
+    public function testValidateWithValidContent(): void
+    {
+        $contentId = 2;
+        $allowedContentTypes = ['article'];
+
+        $this->setupContentTypeValidation($contentId);
+
+        $validationError = $this->targetContentValidator->validate($contentId, $allowedContentTypes);
+
+        self::assertNull($validationError);
+    }
+
+    public function testValidateWithInvalidContentType(): void
+    {
+        $contentId = 2;
+        $allowedContentTypes = ['folder'];
+
+        $this->setupContentTypeValidation($contentId);
+
+        $validationError = $this->targetContentValidator->validate($contentId, $allowedContentTypes);
+
+        self::assertInstanceOf(ValidationError::class, $validationError);
+    }
+
+    private function setupContentTypeValidation(int $contentId): void
+    {
+        $contentTypeId = 55;
+        $contentInfo = new ContentInfo(['id' => $contentId, 'contentTypeId' => $contentTypeId]);
+        $contentType = new ContentType(['identifier' => 'article']);
+
+        $this->contentService
+            ->expects($this->once())
+            ->method('loadContentInfo')
+            ->with($contentId)
+            ->willReturn($contentInfo);
+
+        $this->contentTypeService
+            ->expects($this->once())
+            ->method('loadContentType')
+            ->with($contentInfo->contentTypeId)
+            ->willReturn($contentType);
+    }
+
+    /**
+     * @param mixed $id
+     *
+     * @dataProvider providerForInvalidContentIdentifiers
+     */
+    public function testValidateWithInvalidContentId($id): void
+    {
+        $this->contentService
+            ->expects($this->once())
+            ->method('loadContentInfo')
+            ->with((int) $id)
+            ->willThrowException($this->createMock(NotFoundException::class));
+
+        $validationError = $this->targetContentValidator->validate((int) $id);
+
+        self::assertInstanceOf(ValidationError::class, $validationError);
+    }
+
+    public function providerForInvalidContentIdentifiers(): array
+    {
+        return [
+            ['/foo/bar'],
+            ['test'],
+            ['5'],
+            [[]],
+        ];
+    }
```

Comment:
You're casting those values into `int` anyway, so you can just use the resulting values instead (and strong type `int` in related tests).

Reply:
Wouldn't those values represent the potential invalid arguments provided as a content id? Not sure what should I change here.

Reply:
Nvm, I've refactored this test.